### DISCUSSION TITLE
run the vpn.sh from within a docker

### DISCUSSION THREAD

**Discussion by ronichahwan** [ruyrybeyro/chrootvpn#25] [General]
Hello, 
has anywone tried running this from within a docker 
would it be technically feasible because of the chroot and permission ?

**Comment by ruyrybeyro** [comment]
Considering the current setup of the chroot environment, in conjunction with the inclusion of -cap-add=NET_ADMIN, it is certainly feasible.

 However, it's important to note that despite its capabilities, the container environment will be limited in not being able to manipulate the host routing table (without a dynamic routing protocol) and not changing resolv.conf. These limitations stems from a greater resource isolation. As a result, its functionality will be restricted solely to that of being a router (for the remote network if a split tunnel). 

The decision to employ the chroot format instead of a container was made purposefully, with careful consideration of its specific advantages and suitability for the task at hand.

**Comment by ruyrybeyro** [comment]
TLDR it is possible, it will be a more complicated setup needing a static route or a dynamic routing daemon in the host and inside the container, however resolv.conf in the host wont be changed.
